wrappers/python/simtk/openmm/app/amberprmtopfile.py

```diff
@@ -248,4 +254,16 @@ def createSystem(self, nonbondedMethod=ff.NoCutoff, nonbondedCutoff=1.0*unit.nan
                 force.setEwaldErrorTolerance(ewaldErrorTolerance)
         if removeCMMotion:
             sys.addForce(mm.CMMotionRemover())
+
+        if switchDistance and nonbondedMethod is not ff.NoCutoff:
+            # make sure it's legal
+            if switchDistance >= nonbondedCutoff:
+                raise ValueError('switchDistance is too large compared '
+                                     'to the cutoff!')
+                if abs(switchDistance) != switchDistance:
```

Comment:
No apologies necessary, I've been writing python for all of a week myself!
On 8 Oct 2015 20:48, "Jason Swails" <notifications@github.com> wrote:

> In wrappers/python/simtk/openmm/app/amberprmtopfile.py
> <https://github.com/pandegroup/openmm/pull/1177#discussion_r41552427>:
>
> > @@ -248,4 +254,16 @@ def createSystem(self, nonbondedMethod=ff.NoCutoff, nonbondedCutoff=1.0*unit.nan
> >                  force.setEwaldErrorTolerance(ewaldErrorTolerance)
> >          if removeCMMotion:
> >              sys.addForce(mm.CMMotionRemover())
> > +
> > +        if switchDistance and nonbondedMethod is not ff.NoCutoff:
> > +            # make sure it's legal
> > +            if switchDistance >= nonbondedCutoff:
> > +                raise ValueError('switchDistance is too large compared '
> > +                                     'to the cutoff!')
> > +                if abs(switchDistance) != switchDistance:
>
> Ah, thanks for the pointer. I'll clean up that code a bit (it was some of
> the first OpenMM code I wrote... :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pandegroup/openmm/pull/1177/files#r41552427>.
>